pyproject.toml

```diff
@@ -88,15 +88,15 @@ requires-python = ">=3.11, <3.14"
 [project.optional-dependencies]
 R = ['rpy2 >=3.3']
 extra = [
-  'p_group_cohomology >=3.3',                                                         # Only used in tests
+  #'p_group_cohomology >=3.3', # Only used in tests, not available on PyPI
   'pycosat >=0.6.3',
   'pynormaliz >=2.18; platform_machine != "aarch64" and platform_machine != "arm64"', # Not yet available for Linux aarch64
   'igraph',
   'sage_numerical_backends_coin',
   'symengine >= 0.6.1',                                                               # Only used in tests
   'lrcalc ~=2.1; sys_platform != "win32"',
 ]
-giac = ['sagemath_giac']
+#giac = ['sagemath_giac'] # Not yet available on PyPI (https://github.com/sagemath/sagemath-giac/issues/3)
```

Comment:
wait a minute. Why is "not available on pypi" a reason to not include the package in optional dependencies?

I mean, if the user just happen to already install it manually from source, then pip will be happy anyway?

Reply:
Because uv (and other tools) have no means to install this group automatically, eg you cannot say `pip install sagemath[giac]`.

So there is little point in having it.

Reply:
* if the user types `pip install sagemath[giac]`, or otherwise install a package that requires `sagemath[giac]`:
   * if the user already have `sagemath-giac` installed, doesn't matter
   * if the user doesn't already have it:
      * if the line is included in `pyproject.toml`, the user would get a mysterious error message **at installation**,
      * if the line is excluded from `pyproject.toml`, the user would get a mysterious error message **at runtime**.

I think the former is better.

Reply:
It begs for a separation of the stuff used for CI and the stuff used for "real" installations. Some projects even keep CI in separate repos...